.github/workflows/podvm_mkosi.yaml

```diff
@@ -3,90 +3,209 @@ name: Create a Pod VM image with mkosi
-          push: true
-          build-args:
-            "BUILDER_IMG=ghcr.io/${{ github.repository }}/podvm/builder-fedora:${{ github.sha }}"
+        id: build_binaries
+        working-directory: src/cloud-api-adaptor/podvm-mkosi
+        run: |
+          # If the binaries which matching git sha exists, then skip this build
+          binaries_image=${{ inputs.registry }}/podvm-binaries-fedora-${{ inputs.arch }}
+          tag="$(git rev-parse --short HEAD)"
+          echo "image_tag=${image_tag}" | tee -a "$GITHUB_OUTPUT"
 
-  build-image:
-    name: Build image
-    needs: [build-binaries]
-    if: |
-      always() && (
-          needs.build-binaries.result == 'success' || (
-              needs.build-binaries.result == 'skipped' &&
-              github.event.inputs.binaries-image != ''
-          )
-      )
-    runs-on: ubuntu-24.04
-    permissions:
-      contents: read
-      packages: write
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v4
+          if ! docker manifest inspect "${binaries_image}:${tag}"; then
+            PODVM_BINARIES_IMAGE="${binaries_image}:${tag}" make binaries
+          fi
+          echo "image=${binaries_image}:${tag}" | tee -a "$GITHUB_OUTPUT"
+
+          # If the input has a different image-tag then also push it with that tag
+          if [ -n "${{ inputs.image_tag }}" ] && [ "${{ inputs.image_tag }}" != "${tag}" ];then
+            docker pull "${binaries_image}:${tag}"
+            docker tag "${binaries_image}:${tag}" ${binaries_image}:${{ inputs.image_tag }}
+            docker push "${binaries_image}:${{ inputs.image_tag }}"
+          fi
+        env:
+          PUSH: true
+          ARCH: ${{ inputs.arch }}
+          PODVM_BUILDER_IMAGE: ${{ steps.build_builder.outputs.image }}
+
+      - name: Install mkosi
+        if: ${{ inputs.arch == 's390x' }}
+        run: |
+          git clone -b v22 https://github.com/systemd/mkosi
+          sudo rm -f /usr/local/bin/mkosi
+          sudo ln -s "$PWD/mkosi/bin/mkosi" /usr/local/bin/mkosi
+          mkosi --version
 
       - name: Install Nix
+        if: ${{ inputs.arch == 'amd64' }}
         uses: cachix/install-nix-action@v30
 
       - name: Build nix shell to cache dependencies
+        if: ${{ inputs.arch == 'amd64' }}
         run: nix build .#devShells.x86_64-linux.podvm-mkosi
 
-      - name: Install crane
+      - name: Install crane with nix
+        if: ${{ inputs.arch == 'amd64' }}
         run: nix profile install nixpkgs#crane
 
+      - name: Install crane with go
+        if: ${{ inputs.arch == 's390x' }}
+        run: |
+          go install github.com/google/go-containerregistry/cmd/crane@latest
+          echo "PATH=${PATH}:$(go env GOPATH)/bin" >> "$GITHUB_ENV"
+
         # This removes the checkout and creates a btrfs volume with maximized
         # build space.
       - name: Maximize build space
+        if: ${{ inputs.arch == 'amd64' }}
```

Comment:
I don't think we need this step anymore, do we?

Reply:
I'm not sure - on the github runner we did run out of space a lot in the past. I will try without this step tomorrow though.